auth/api/iam/session.go

```diff
@@ -39,17 +39,76 @@ type OAuthSession struct {
 	ClientState            string
 	SessionID              string
 	RedirectURI            string
-	ServerState            ServerState
 	ResponseType           string
 	PKCEParams             PKCEParams
 	PresentationDefinition PresentationDefinition
 	VerifierDID            *did.DID
+	OpenID4VPVerifier      *OpenID4VPVerifier
+}
 
-	// TODO use these 2 fields to track if all OpenID4VP flows have been concluded
-	// PresentationSubmissions tracks which PresentationSubmissions have been submitted through OpenID4VP
-	PresentationSubmissions map[string]pe.PresentationSubmission
+// OpenID4VPVerifier tracks the verifier's state of multiple OpenID4VP flows that all relate to a single OAuthSession (e.g. Authorization Code flow).
+// It's a sort of state machine with 2 states.
+type OpenID4VPVerifier struct {
+	WalletDID                       did.DID
+	RequiredPresentationDefinitions pe.WalletOwnerMapping
+	// Submissions tracks which Submissions have been submitted through OpenID4VP
+	Submissions map[string]pe.PresentationSubmission
 	// Presentations tracks which VerifiablePresentations have been received through OpenID4VP
-	Presentations map[string]vc.VerifiablePresentation
+	Presentations []vc.VerifiablePresentation
```

Comment:
Submissions is a map, this is a slice. Shouldn't submissions and presentations be stored in pairs?

Reply:
Probably, yes. And `Credentials` be removed, because it's the output of `resolve(presentation, submission, definition)`